src/module.ts

```diff
@@ -75,7 +94,10 @@ const defaults: ModuleOptions & { basePath: string } = {
   trustHost: false,
   enableSessionRefreshPeriodically: false,
   enableSessionRefreshOnWindowFocus: true,
-  enableGlobalAppMiddleware: false
+  enableGlobalAppMiddleware: false,
```

Comment:
Maybe we can merge `enableGlobalAppMiddleware` with `globalMiddlewareOptions`, so we have all in one place. What do you think?

Something like this:
```ts
globalAppMiddleware: {
  enabled: false,
  allow404WithoutAuth: true
}
```

Reply:
I like the idea very much, I think it could even have a signature of:
```
globalAppMiddleware: boolean | GlobalAppMiddlewareOptions
```

with `false` turning it off, `true` turning it on with default config and `{ ... }` allowing for granular configuration. 

An issue with this approach and why I didn't go with it so far:
- breaks backwards-compat (less severe if we go with the `boolean | ...` signature)
- maybe does not work with the way that `nuxt/schema` generates the typings for the config keys -> there's some really weird errors there

I would look into whether I can mitigate the two problems and add it here, otherwise I propose that we move it to the future, maybe also as part of a `0.4.x` jump in versioning (incoming anyways, due to requirement to make `next-auth ` a peer-dep which is definitely a breaking change, see #128 for example)

Reply:
I liked the signature you suggested! Agreed.

One thing about versioning, maybe we should warn users to fix the exact version they are using to avoid breaking changes until we release v1.0.0

Reply:
We're already following the policy of only releasing:
- patch changes that are non-breaking.
- minor changes if something breaking happened 

Do you think that's enough to avoid what you described?

Reply:
Yeah, based on what you said on discord about the approach of using `0.x` for now, I guess so.
But I think we could suggest users to add `nuxt-auth` using `~0.x` instead of `^0.x` in `package.json`, so it will only update to future patch versions, without incrementing the minor version.